tests/Fixtures/FirstClassCollection.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace JMS\Serializer\Tests\Fixtures;
+
+use JMS\Serializer\Annotation as Serializer;
+
+class FirstClassCollection implements \IteratorAggregate
+{
+    /**
+     * @Serializer\Type("array<int>")
+     * @Serializer\Inline
```

Comment:
Currently, your first test case works just by coincidence. `@Inline` was never tested on collections.
If will also fail if your you unset one item from the collection (making keys not consequent).

And on top of it what it the expected result if you add one more property to the  `FirstClassCollection` ?

Reply:
> what it the expected result if you add one more property

Depends. Either an error as there could only be one property with ``@Inline, or, if you consider things like exclusion strategy, error of there is more than one property after applying exclusion rules.

Reply:
An error (doable at "metadata" discovery time) should be fine

Reply:
Consider that the collection could be also an associative array, so it can be both `[]` and `{}`